pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -115,30 +119,35 @@ func ConvertService(svc coreV1.Service, domainSuffix string, clusterID string) *
 		},
 	}
 
-	if svc.Spec.Type == coreV1.ServiceTypeNodePort {
+	switch svc.Spec.Type {
+	case coreV1.ServiceTypeNodePort:
+		if svc.Annotations[NodeSelectorAnnotation] == "" {
+			// only do this for istio ingress-gateway services
```

Comment:
How can the user specify pick all nodes?

Reply:
got it, though it is not recommended to do use it like this.